src/PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "PointwiseFunctions/AnalyticData/CurvedWaveEquation/PureSphericalHarmonic.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/EagerMath/Magnitude.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Evolution/Systems/CurvedScalarWave/Tags.hpp"
+#include "NumericalAlgorithms/Spectral/SwshInterpolation.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace CurvedScalarWave::AnalyticData {
+
+PureSphericalHarmonic::PureSphericalHarmonic(
+    double radius, const double width, const size_t l, const int m,
+    const Options::Context& /*context*/)
+    : radius_(radius), width_sq_(width * width), l_mode_(l), m_mode_(m) {
+  // LCOV_EXCL_START
+  if (abs(m_mode_) > static_cast<int>(l_mode_)) {
+    ERROR(
+        "The absolute value of the m_mode must be less than or equal to the "
+        "l-mode but the m-mode is "
+        << m_mode_ << " and the l-mode is" << l_mode_);
+  }
+  if (radius <= 0.) {
+    ERROR("The radius must be greater than 0 but is " << radius_);
+  }
+  if (width <= 0.) {
+    ERROR("The width must be greater than 0 but is " << width);
+  }
+  // LCOV_EXCL_STOP
+}
+
+tuples::TaggedTuple<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                    CurvedScalarWave::Psi>
+PureSphericalHarmonic::variables(
+    const tnsr::I<DataVector, 3>& x, double /*t*/,
+    tmpl::list<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+               CurvedScalarWave::Psi> /*meta*/) const noexcept {
+  Scalar<DataVector> pi{get(magnitude(x)) - radius_};
+  get(pi) = exp(-get(pi) * get(pi) / width_sq_);
+  const Spectral::Swsh::SpinWeightedSphericalHarmonic spherical_harmonic(
+      0, l_mode_, m_mode_);
+  const auto theta = atan2(hypot(x[0], x[1]), x[2]);
+  const auto phi = atan2(x[1], x[0]);
+  get(pi) *= real(spherical_harmonic.evaluate(theta, phi, sin(theta / 2.),
+                                              cos(theta / 2.)));
+  return tuples::TaggedTuple<CurvedScalarWave::Pi, CurvedScalarWave::Phi<3>,
+                             CurvedScalarWave::Psi>{
+      std::move(pi), make_with_value<CurvedScalarWave::Phi<3>::type>(x, 0.),
+      make_with_value<CurvedScalarWave::Psi::type>(pi, 0.)};
```

Comment:
just saw this, which I will change since `pi` might be moved from...

Reply:
Just use `x` for the size. While you're at it, just use explicit tensor types instead of the `::type`, it's more readable.